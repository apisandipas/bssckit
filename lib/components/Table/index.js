"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Table", {
  enumerable: true,
  get: function get() {
    return _Table["default"];
  }
});
Object.defineProperty(exports, "Tr", {
  enumerable: true,
  get: function get() {
    return _Tr["default"];
  }
});

var _Table = _interopRequireDefault(require("./Table"));

var _Tr = _interopRequireDefault(require("./Tr"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }