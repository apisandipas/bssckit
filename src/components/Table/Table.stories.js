import React from 'react';
import { Table, Tr } from './index';

export default {
  title: 'Table',
};

const Thead = () => (
  <thead>
    <tr>
      <th scope='col'>#</th>
      <th scope='col'>First</th>
      <th scope='col'>Last</th>
      <th scope='col'>Handle</th>
    </tr>
  </thead>
);

const Tbody = () => (
  <tbody>
    <tr>
      <th scope='row'>1</th>
      <td>Mark</td>
      <td>Otto</td>
      <td>@mdo</td>
    </tr>
    <tr>
      <th scope='row'>2</th>
      <td>Jacob</td>
      <td>Thornton</td>
      <td>@fat</td>
    </tr>
    <tr>
      <th scope='row'>3</th>
      <td>Larry</td>
      <td>the Bird</td>
      <td>@twitter</td>
    </tr>
  </tbody>
);

export const simple = () => (
  <Table>
    <Thead />
    <Tbody />
  </Table>
);

export const dark = () => (
  <Table tableDark>
    <Thead />
    <Tbody />
  </Table>
);

export const darkThead = () => (
  <Table theadDark>
    <Thead />
    <Tbody />
  </Table>
);

export const lightThead = () => (
  <Table theadLight>
    <Thead />
    <Tbody />
  </Table>
);

export const stripped = () => (
  <Table striped>
    <Thead />
    <Tbody />
  </Table>
);

export const strippedDark = () => (
  <Table striped tableDark>
    <Thead />
    <Tbody />
  </Table>
);

export const bordered = () => (
  <Table bordered>
    <Thead />
    <Tbody />
  </Table>
);

export const hoverable = () => (
  <Table hover>
    <Thead />
    <Tbody />
  </Table>
);

export const hoverableDark = () => (
  <Table hover tableDark>
    <Thead />
    <Tbody />
  </Table>
);

export const small = () => (
  <Table sm>
    <Thead />
    <Tbody />
  </Table>
);

export const smallDark = () => (
  <Table sm tableDark>
    <Thead />
    <Tbody />
  </Table>
);

export const contextual = () => (
  <Table>
    <thead>
      <tr>
        <th scope='col'>Color</th>
        <th scope='col'>Content</th>
      </tr>
    </thead>
    <tbody>
      <Tr active>
        <td>Active</td>
        <td>Content</td>
      </Tr>
      <Tr primary>
        <td>Primary</td>
        <td>Content</td>
      </Tr>
      <Tr secondary>
        <td>Secondary</td>
        <td>Content</td>
      </Tr>
      <Tr success>
        <td>Success</td>
        <td>Content</td>
      </Tr>
      <Tr danger>
        <td>Danger</td>
        <td>Content</td>
      </Tr>
      <Tr warning>
        <td>Warning</td>
        <td>Content</td>
      </Tr>
      <Tr info>
        <td>Info</td>
        <td>Content</td>
      </Tr>
      <Tr light>
        <td>Light</td>
        <td>Content</td>
      </Tr>
      <Tr dark>
        <td>Dark</td>
        <td>Content</td>
      </Tr>
    </tbody>
  </Table>
);

export const caption = () => (
  <Table>
    <caption>List of users</caption>
    <Thead />
    <Tbody />
  </Table>
);

export const responsive = () => (
  <Table responsive>
    <Thead />
    <Tbody />
  </Table>
);

export const responseAtBreakpoints = () => (
  <Table responsiveMd>
    <Thead />
    <Tbody />
  </Table>
);
