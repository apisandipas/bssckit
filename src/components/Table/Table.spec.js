import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import Table from './Table';

describe('Table component', () => {
  it('matches Table', () => {
    const { container } = render(<Table />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches Table bordered', () => {
    const { container } = render(<Table bordered />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches Table hover', () => {
    const { container } = render(<Table hover />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches Table responsive bordered', () => {
    const { container } = render(<Table responsive bordered />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'block');
    expect(container.firstChild).toHaveStyleRule('width', '100%');
    expect(container.firstChild).toHaveStyleRule('overflow-x', 'auto');
  });

  it('matches Table responsiveLg bordered', () => {
    const { container } = render(<Table responsiveLg bordered />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'block', {
      media: '(max-width:991.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('width', '100%', {
      media: '(max-width:991.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('overflow-x', 'auto', {
      media: '(max-width:991.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('border', '0', {
      media: '(max-width:991.98px)',
    });
  });

  it('matches Table responsiveMd bordered', () => {
    const { container } = render(<Table responsiveMd bordered />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'block', {
      media: '(max-width:767.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('width', '100%', {
      media: '(max-width:767.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('overflow-x', 'auto', {
      media: '(max-width:767.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('border', '0', {
      media: '(max-width:767.98px)',
    });
  });

  it('matches Table responseSm bordered', () => {
    const { container } = render(<Table responsiveSm bordered />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'block', {
      media: '(max-width:575.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('width', '100%', {
      media: '(max-width:575.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('overflow-x', 'auto', {
      media: '(max-width:575.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('border', '0', {
      media: '(max-width:575.98px)',
    });
  });

  it('matched Table responsiveXl bordered', () => {
    const { container } = render(<Table responsiveXl bordered />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'block', {
      media: '(max-width:1199.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('width', '100%', {
      media: '(max-width:1199.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('overflow-x', 'auto', {
      media: '(max-width:1199.98px)',
    });
    expect(container.firstChild).toHaveStyleRule('border', '0', {
      media: '(max-width:1199.98px)',
    });
  });

  it('matches Table small', () => {
    const { container } = render(<Table sm />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('padding', '0.3rem', {
      modifier: '& th',
    });
    expect(container.firstChild).toHaveStyleRule('padding', '0.3rem', {
      modifier: '& td',
    });
  });

  it('matches Tabled striped', () => {
    const { container } = render(<Table stiped />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches Table tableDark bordered striped hover', () => {
    const { container } = render(<Table tableDark bordered striped hover />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches Table theadDark', () => {
    const { container } = render(<Table theadDark />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches Table theadLight', () => {
    const { container } = render(<Table theadLight />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
