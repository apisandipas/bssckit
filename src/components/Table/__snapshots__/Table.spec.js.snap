// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component matched Table responsiveXl bordered 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
  border: 1px solid #e2e6ea;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 th,
.c1 td {
  border: 1px solid #e2e6ea;
}

.c1 thead th,
.c1 thead td {
  border-bottom-width: 2px;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (max-width:1199.98px) {
  .c1 {
    display: block;
    width: 100%;
    overflow-x: auto;
    border: 0;
  }
}

<table
  class="sc-fzoLag c0 c1"
/>
`;

exports[`Table component matches Table 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<table
  class="sc-fzoLag c0 c1"
/>
`;

exports[`Table component matches Table bordered 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
  border: 1px solid #e2e6ea;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 th,
.c1 td {
  border: 1px solid #e2e6ea;
}

.c1 thead th,
.c1 thead td {
  border-bottom-width: 2px;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<table
  class="sc-fzoLag c0 c1"
/>
`;

exports[`Table component matches Table hover 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 tbody tr:hover {
  background-color: rgba(0,0,0,0.25);
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<table
  class="sc-fzoLag c0 c1 hover"
/>
`;

exports[`Table component matches Table responseSm bordered 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
  border: 1px solid #e2e6ea;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 th,
.c1 td {
  border: 1px solid #e2e6ea;
}

.c1 thead th,
.c1 thead td {
  border-bottom-width: 2px;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (max-width:575.98px) {
  .c1 {
    display: block;
    width: 100%;
    overflow-x: auto;
    border: 0;
  }
}

<table
  class="sc-fzoLag c0 c1"
/>
`;

exports[`Table component matches Table responsive bordered 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
  border: 1px solid #e2e6ea;
  display: block;
  width: 100%;
  overflow-x: auto;
  border: 0;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 th,
.c1 td {
  border: 1px solid #e2e6ea;
}

.c1 thead th,
.c1 thead td {
  border-bottom-width: 2px;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<table
  class="sc-fzoLag c0 c1"
/>
`;

exports[`Table component matches Table responsiveLg bordered 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
  border: 1px solid #e2e6ea;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 th,
.c1 td {
  border: 1px solid #e2e6ea;
}

.c1 thead th,
.c1 thead td {
  border-bottom-width: 2px;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (max-width:991.98px) {
  .c1 {
    display: block;
    width: 100%;
    overflow-x: auto;
    border: 0;
  }
}

<table
  class="sc-fzoLag c0 c1"
/>
`;

exports[`Table component matches Table responsiveMd bordered 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
  border: 1px solid #e2e6ea;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 th,
.c1 td {
  border: 1px solid #e2e6ea;
}

.c1 thead th,
.c1 thead td {
  border-bottom-width: 2px;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

@media (max-width:767.98px) {
  .c1 {
    display: block;
    width: 100%;
    overflow-x: auto;
    border: 0;
  }
}

<table
  class="sc-fzoLag c0 c1"
/>
`;

exports[`Table component matches Table small 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 th,
.c1 td {
  padding: 0.3rem;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<table
  class="sc-fzoLag c0 c1"
/>
`;

exports[`Table component matches Table tableDark bordered striped hover 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
  border: 1px solid #e2e6ea;
  color: #fff;
  background-color: #2d2f31;
  border: 0;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 th,
.c1 td {
  border: 1px solid #e2e6ea;
}

.c1 thead th,
.c1 thead td {
  border-bottom-width: 2px;
}

.c1 tbody tr:nth-of-type(odd) {
  background-color: rgba(0,0,0,0.1);
}

.c1 tbody tr:hover {
  background-color: rgba(0,0,0,0.25);
}

.c1 th,
.c1 td,
.c1 thead th {
  border-color: #212529;
}

.c1 tbody tr:nth-of-type(odd) {
  background-color: rgba(255,255,255,0.1);
}

.hover .c1 tbody tr:hover {
  background-color: rgba(255,255,255,0.1);
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<table
  class="sc-fzoLag c0 c1 hover"
/>
`;

exports[`Table component matches Table theadDark 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 thead th {
  color: #fff;
  background-color: #2d2f31;
  border-color: #212529;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<table
  class="sc-fzoLag c0 c1"
/>
`;

exports[`Table component matches Table theadLight 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

.c1 thead th {
  color: #343a40;
  background-color: #d6d8d9;
  border-color: #d6d8d9;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<table
  class="sc-fzoLag c0 c1"
/>
`;

exports[`Table component matches Tabled striped 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  background-color: transparent;
}

.c1 th {
  text-align: inherit;
}

.c1 th,
.c1 td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #d6d8d9;
}

.c1 thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #d6d8d9;
}

.c1 tbody + tbody {
  border-top: 2px solid #d6d8d9;
}

.c1 + .c0 {
  background-color: #fff;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<table
  class="sc-fzoLag c0 c1"
/>
`;
